autoload.php

```diff
@@ -1,6 +1,6 @@
 <?php
 
-defined('WPINC') || die;
```

Comment:
`ABSPATH` defines the absolute path to the WordPress directory, while `WPINC` defines the location of the WordPress directory of functions, classes, and core content. Is there a specific reason why you would choose `ABSPATH` over `WPINC`?

Reply:
This is simple: WordPress uses ABSPATH to check core way loaded (zillion times) -> the WordPress extension does the same -> testable code should do the same 🐈

Reply:
`defined('WPINC') || die;` makes the analysis stop as `autoload.php` is a bootstrap file

Reply:
Ah, so you mean it is only useful to change in the autoload file? This is called during the plugins_loaded hook by which time `WPINC` has already been defined. So does it make a difference?

Reply:
Paul, as core always checks ABSPATH every code for WordPress should - although many-many plugin goes for WPINC

Please search the core for ABSPATH checking and see for yourself :)

Reply:
```
wp-admin/edit-form-blocks.php:12:if ( ! defined( 'ABSPATH' ) ) {
wp-admin/link-parse-opml.php:9:if ( ! defined( 'ABSPATH' ) ) {
wp-admin/admin-post.php:16:if ( defined( 'ABSPATH' ) ) {
wp-admin/edit-form-comment.php:10:if ( ! defined( 'ABSPATH' ) ) {
wp-admin/edit-link-form.php:10:if ( ! defined( 'ABSPATH' ) ) {
wp-admin/load-styles.php:11:if ( ! defined( 'ABSPATH' ) ) {
wp-admin/async-upload.php:17:if ( defined( 'ABSPATH' ) ) {
wp-admin/load-scripts.php:11:if ( ! defined( 'ABSPATH' ) ) {
wp-admin/admin-footer.php:10:if ( ! defined( 'ABSPATH' ) ) {
wp-admin/edit-form-advanced.php:10:if ( ! defined( 'ABSPATH' ) ) {
wp-admin/edit-tag-form.php:10:if ( ! defined( 'ABSPATH' ) ) {
wp-admin/setup-config.php:29:if ( ! defined( 'ABSPATH' ) ) {
wp-config-sample.php:85:if ( ! defined( 'ABSPATH' ) ) {
wp-cron.php:42:if ( ! defined( 'ABSPATH' ) ) {
wp-load.php:20:if ( ! defined( 'ABSPATH' ) ) {
```